src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -378,25 +389,90 @@ def initRun(self, run_context: "RunContext", parameters: List[str] = None):
                         )
+                        or current_status == STATE_LOAD_FAILURE
+                    ):
+                        rng = self.realizations[realization_nr]
+                        impl_type = enkf_node.getImplType()
+
+                        if impl_type == ErtImplType.GEN_KW:
+                            gen_kw_node = enkf_node.asGenKw()
+                            file = self._alloc_initfile(config_node, realization_nr)
+
+                            if file is not None and file.exists():
+                                gen_kw_in_config = [p[0] for p in gen_kw_node.items()]
+                                vals = self._load_from_file_gen_kw(
+                                    gen_kw_in_config, file
+                                )
+                            else:
+                                vals = self._sample_gen_kw(rng, len(gen_kw_node))
+
+                            if len(vals) > 0:
+                                s = io.BytesIO()
+                                # The first element is time_t (64 bit integer), but it is
+                                # not used so we write 0 instead - for padding purposes.
+                                s.write(struct.pack("Qi", 0, int(ErtImplType.GEN_KW)))
+                                s.write(vals.tobytes())
+
+                                enkf_fs.write_parameter(
+                                    run_context.sim_fs,
+                                    config_node.getKey(),
+                                    realization_nr,
+                                    s.getvalue(),
+                                )
+
+                        else:
+                            enkf_state.state_initialize(
+                                self.realizations[realization_nr],
+                                enkf_node,
+                                run_context.sim_fs,
+                                realization_nr,
+                            )
+
                 if state_map[realization_nr] in [STATE_UNDEFINED, STATE_LOAD_FAILURE]:
                     state_map[realization_nr] = RealizationStateEnum.STATE_INITIALIZED
         run_context.sim_fs.sync()
 
+    def _alloc_initfile(self, config_node: EnkfConfigNode, iens: int) -> Optional[Path]:
+        file_format = config_node.get_init_file_fmt()
+
+        if not file_format:
+            return None
+
+        return Path(
+            file_format % iens if "%d" in file_format else file_format
+        ).absolute()
+
+    def _sample_gen_kw(
+        self, rng: RandomNumberGenerator, nsamples: int
+    ) -> npt.NDArray[np.float_]:
+        return np.array([_clib.enkf_util.standard_normal(rng) for _ in range(nsamples)])
+
+    def _load_from_file_gen_kw(
+        self, gen_kw: List[str], file: Optional[Path]
+    ) -> npt.NDArray[np.float_]:
+        if file is not None and file.exists():
+            df = pd.read_csv(file, sep=" ", header=None)
+            if df.shape[1] == 2:
```

Comment:
The config node object stores the keys in a specific order. When reading in as tuples in C++, the order of the keys is not required to be the same. In this approach  the order would have to be the same to assign them to correct parameter key.